core/shared/src/main/scala/org/typelevel/idna4s/core/uts46/CodePointMapper.scala

```diff
@@ -0,0 +1,220 @@
+package org.typelevel.idna4s.core.uts46
+
+import scala.annotation.tailrec
+import scala.annotation.switch
+import cats.data._
+import org.typelevel.idna4s.core._
+import scala.util.control.NoStackTrace
+import java.lang.StringBuilder
+
+private[uts46] trait CodePointMapperBase {
+  protected final val VALID: Int = -1
+  protected final val VALID_NV8: Int = -2
+  protected final val VALID_XV8: Int = -3
+  protected final val IGNORED: Int = -4
```

Comment:
Same for all of these: I think if we can drop the `: Int` there's a good chance these values will be inlined. But since this is in a `trait` we'd have to check the bytecode to be sure 😅 

Also, can this be an `abstract class` instead?

Reply:
One more option is you could declare these in `CodePointMapper`, and then in `GeneratedCodePointMapper` do a self type `this: CodePointMapper.type =>`.

Reply:
Dropping the type ascription fixed the inlining. Thanks!